tests/backup/backup_restore_basic_test.go

```diff
@@ -251,25 +253,64 @@ var _ = Describe("{CustomResourceBackupAndRestore}", func() {
 			ctx, err := backup.GetAdminCtxFromSecret()
 			log.FailOnError(err, "Fetching px-central-admin ctx")
 			for _, backupName := range backupNames {
-				restoreName = fmt.Sprintf("%s-%s-%v", restoreNamePrefix, backupName, time.Now().Unix())
+				restoreName = fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
 				restoreNames = append(restoreNames, restoreName)
 				namespaceList, err := FetchNamespacesFromBackup(ctx, backupName, orgID)
 				for _, namespace := range namespaceList {
-					restoredNameSpace := fmt.Sprintf("%s-%s-%s", backupNames, "restored", namespace)
+					restoredNameSpace := fmt.Sprintf("%s-%s", RandomString(10), "restored")
 					namespaceMapping[namespace] = restoredNameSpace
 				}
+				log.InfoD("Namespace mapping is %v:", namespaceMapping)
+				restoreNamespaceMappingMap[restoreName] = namespaceMapping
 				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, namespaceList)
-				err = CreateRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), SourceClusterName, orgID, appContextsToBackup)
+				restoreContextMap[restoreName] = appContextsToBackup
+				err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, orgID, ctx, make(map[string]string))
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore: %s from backup: %s", restoreName, backupName))
 			}
 		})
 
+		Step("Validating custom resource backup restores", func() {
```

Comment:
@snigdha-px Can we make generic library ? Cc: @mkoppal-px @atrivedi-px

Reply:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3896/

Reply:
done